lua/autorun/client/cl_roundinfo_handler.lua

```diff
@@ -166,7 +166,7 @@ if CLIENT then
     function KILLER_INFO:PrintSelf()
         if (GAMEMODE.round_state ~= ROUND_ACTIVE and GAMEMODE.round_state ~= ROUND_POST) then return end
 
-        local txt = LANG.GetParamTranslation('ttt_rs_suicideText', {killtype = self.data.damage_type_name})
+        local txt = LANG.GetParamTranslation('ttt_rs_suicideText', {killtype = self.data.killer_weapon_name != 'WEAPON_NAME' and self.data.killer_weapon_name or self.data.damage_type_name})
```

Comment:
I would suggest to rework this, since `self.data.killer_weapon_name != 'WEAPON_NAME'` will misbehave, when a player already got killed by a weapon, as the self.data.killer_weapon_name is never reset to `WEAPON_NAME`

Reply:
I didn't realize that the killer info is reused. Instead of reworking said line of code, I think it would make more sense to make sure that the shared object is set back to its initial values during a call to "HidePopup". That way the table would be cleaned up before it is used again. I think it makes sense to get rid of the leftovers from the last time it was used. What do you think?

Reply:
I implemented the change I proposed and the issue should be resolved. Please let me know what you think.